### DISCUSSION TITLE
1000s of .tmp files on s3 remote, maybe contributing to slow response?

### DISCUSSION THREAD

**Discussion by wdixon** [treeverse/dvc#6802] [Help]
The internal minio sever we are using to host the remote repository has 10000's of .tmp files sitting at the same level as the 00-FF directories that hold the git style hashed repository....   We are starting to see some slow responses and there is an indication that it has to do with the number of these files.

The question is, should these be there?  And if not - what are they from?  are they safe to remove?

**Comment by efiop** [comment]
Hi @wdixon ! We use those for uploading files, so that they then could be moved into place in one operation. The fact that you have so many of them, means that those are probably incomplete uploads that broke at some point. You can safely remove those files.

**Reply by karajan1001** [reply]
Maybe `dvc gc -c` should remove them?
